### PR TITLE
Feature/cypress testing

### PR DESCRIPTION
**Pull request by kevinm23nelson**
## What does this PR do?
This PR adds Cypress tests to ensure that various user flows in the application are functioning correctly. These tests cover scenarios such as navigating through the app, interacting with UI elements, and verifying data rendering on different pages.

## Where should the reviewer start?
The reviewer should start by looking at the Cypress test files located in the `cypress/integration` directory. Reviewing the test files will provide an understanding of the different user flows being tested and the structure of the tests.

## How should this be manually tested?
To manually test the changes, run the Cypress test suite using the command `npx cypress open` or `npx cypress run`. This will launch Cypress and execute the tests, providing visual feedback and logs for each test scenario. Verify that all tests pass and that the application behaves as expected.

## Any background context you want to provide?
These tests were created to ensure the robustness of the application by simulating real user interactions. The tests cover critical user flows, such as adding and removing favorites, viewing currency details, and verifying that the UI updates correctly based on user actions.

## What are the relevant tickets (if any)?
#13 

## Screenshots (if appropriate)
Screenshot 2024-08-01 at 2 19 50 PM

## Questions:
- Are there any additional user flows or edge cases that should be covered by the tests?
- Are the tests comprehensive enough to catch potential regressions in future updates?
- Is the current setup for running and organizing Cypress tests optimal for the team's workflow?

### CODE REVIEW

File: CryptoPal/cypress/e2e/Dashboard_spec.cy.js [hunk 1]
```diff
@@ -1,5 +1,40 @@
...
+        cy.get('.sidebar-button').eq(1).should('contain.text', 'Explore');
+      });
+    });
+  });
+
+  it('displays the dashboard correctly', () => {
+    // Assert that the dashboard header says Dashboard
+    cy.get('.dashboard-header').should('contain.text', 'Dashboard');
+
+    // Assert that there is a filter with the correct options
```
Comment (Sterling47) [inline CryptoPal/cypress/e2e/Dashboard_spec.cy.js]: Nice use of comments to keep testing organized

File: CryptoPal/cypress/support/commands.js [hunk 2]
```diff
@@ -1,19 +1,20 @@
 Cypress.Commands.add('setupCurrencyIntercepts', () => {
-    console.log('Setting up intercepts');
-    
+    // Intercept the request to get all currencies
```
Comment (Sterling47) [inline CryptoPal/cypress/support/commands.js:2]: Nice use of intercept & fixture

File: CryptoPal/cypress/e2e/SearchBar_spec.cy.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
+describe('SearchBar', () => {
+  beforeEach(() => {
+    // Set up intercepts with fixture data
+    cy.setupCurrencyIntercepts();
```
Comment (Sterling47) [inline CryptoPal/cypress/e2e/SearchBar_spec.cy.js:4]: is this missing a fixture?
